File: DSA/IshikaGarg_500124973/Day_17_Q3.c
#include <stdio.h>

int squareRoot(int n) {
    if (n == 0 || n == 1)
        return n;
        
    int left = 1, right = n, ans = 0;
    while (left <= right) {
        int mid = (left + right) / 2;
        if (mid <= n / mid) {
            ans = mid;
            left = mid + 1;
        } else
            right = mid - 1;
    }
    return ans;
}

int main() {
    int n = 27;
    int result = squareRoot(n);
    printf("%d\n", result);  
    return 0;
}

File: DSA/NarindVerma_500121898/Day17Q3.py
def integer_square_root(n):
    if n < 2:
        return n
        
    left, right = 1, n // 2
    result = 0
    
    while left <= right:
        mid = (left + right) // 2
        mid_squared = mid * mid
        
        if mid_squared == n:
            return mid
        elif mid_squared < n:
            result = mid  # store last valid mid
            left = mid + 1
        else:
            right = mid - 1
            
    return result
